File: api/Entities/Deployments/Dep.cs
using System.ComponentModel.DataAnnotations.Schema;
using api.Entities.HR;

namespace api.Entities.Deployments
{
    public class Dep: BaseEntity
    {
        public Dep()
        {
        }

        //[ForeignKey("SelectionDecisionId")]
        //public int SelectionDecisionId { get; set; }
        public int CvRefId { get; set; }
        public int OrderItemId { get; set; }
        public int CustomerId { get; set; }
        public string CustomerName { get; set; }
        public string CityOfWorking { get; set; }
        public DateTime SelectedOn { get; set; }
        public string CurrentStatus { get; set; }
        public DateTime CurrentStatusDate { get; set; }
        public bool Ecnr { get; set; }=false;
        public ICollection<DepItem> DepItems { get; set; }
        public CVRef CVRef { get; set; }

    }


}

File: client/src/app/_models/process/dep.ts
import { IDepItem } from "./depItem";

export interface IDep
{
	id: number;
	cvRefId: number;
	orderItemId: number;
	customerId: number;
	customerName: string;
	cityOfWorking: string;
	selectedOn: Date;
	currentStatus: string;
	depItems: IDepItem[];
	ecnr: boolean;

}


export class Dep implements IDep
{
	id: number=0;
	cvRefId: number=0;
	orderItemId: number=0;
	customerId: number=0;
	customerName: string ='';
	cityOfWorking: string = '';
	selectedOn: Date = new Date();
	currentStatus: string = '';
	depItems: IDepItem[] = [];
	ecnr = false;
}